src/audio/kpb.c

```diff
@@ -628,7 +629,17 @@ static int kpb_reset(struct comp_dev *dev)
 		ret = comp_set_state(dev, COMP_TRIGGER_RESET);
 		break;
 	default:
+		kpb->kpb_no_of_clients = 0;
```

Comment:
I would probably leave the stuff in the event framework untouched as that stuff is all async weirdness, unless you are confident everything gets set back up again

Reply:
Removed.
It's also reset to 0 in prepare function so this reset should be no harm but not needed here.